File: rest/src/main/java/io/sparkled/rest/service/effecttype/EffectTypeRestService.java
package io.sparkled.rest.service.effecttype;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/effectTypes")
public class EffectTypeRestService {

    private final EffectTypeRestServiceHandler handler;

    @Inject
    public EffectTypeRestService(EffectTypeRestServiceHandler handler) {
        this.handler = handler;
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response get() {
        return handler.get();
    }
}

File: rest/src/main/kotlin/io/sparkled/rest/service/effecttype/EffectTypeRestService.kt
package io.sparkled.rest.service.effecttype

import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/effectTypes")
class EffectTypeRestService
@Inject constructor(private val handler: EffectTypeRestServiceHandler) {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun get(): Response {
        return handler.get()
    }
}